core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1007,19 +1007,27 @@ private SqlCall toSql(RexProgram program, RexOver rexOver) {
           orderList, isRows, lowerBound, upperBound, allowPartial, POS);
 
       final List<SqlNode> nodeList = toSql(program, rexOver.getOperands());
-      return createOverCall(sqlAggregateFunction, nodeList, sqlWindow);
+      return createOverCall(sqlAggregateFunction, nodeList, sqlWindow, rexOver.isDistinct());
     }
 
     private SqlCall createOverCall(SqlAggFunction op, List<SqlNode> operands,
-        SqlWindow window) {
+        SqlWindow window, boolean isDistinct) {
       if (op instanceof SqlSumEmptyIsZeroAggFunction) {
         // Rewrite "SUM0(x) OVER w" to "COALESCE(SUM(x) OVER w, 0)"
         final SqlCall node =
-            createOverCall(SqlStdOperatorTable.SUM, operands, window);
+            createOverCall(SqlStdOperatorTable.SUM, operands, window, isDistinct);
```

Comment:
Can we enter here with `isDistinct == true`? If yes can you also add a testcase?

Reply:
Sorry?

Reply:
Seems can enter, I test "SELECT sum(distinct employee_id) over (order by hire_date) FROM employee" can enter, but I can not write a SQL with SUM0, do you have any advice about this, or I can add the SUM DISTINCT query?

Reply:
If I remember well `SUM0` appears when the column is nullable. Did you try an example with a nullable column?

Reply:
ok, I'll take a try